src/extension/index.ts

```diff
@@ -109,6 +111,52 @@ function activateDiagnostics(disposables: Disposable[], store: Store, baser: Uri
     disposables.push(workspace.onDidCloseTextDocument(doc => diagsAll.delete(doc.uri))); // Spurious *.git deletes don't hurt.
     const disposer = observe(store.logs, () => workspace.textDocuments.forEach(setDiags));
     disposables.push({ dispose: disposer });
+    return setDiags;
+}
+
+function activateCodeActions(disposables: Disposable[], baser: UriRebaser, refreshDiagnostics: (document: TextDocument) => void) {
+    const commandId = 'sarif.fix';
+    disposables.push(commands.registerCommand(commandId, async (document: TextDocument, result: Result, fix: Fix) => {
+        const edit = new WorkspaceEdit();
+        for (const change of fix.artifactChanges) {
+            const { artifactLocation, replacements } = change;
+            const [artifactUri] = parseArtifactLocation(result, artifactLocation);
+            if (!artifactUri) continue;
+
+            const localUri = await baser.translateArtifactToLocal(artifactUri);
+            for (const replacement of replacements) {
+                edit.replace(
+                    Uri.parse(localUri),
+                    regionToSelection(document, replacement.deletedRegion),
+                    replacement.insertedContent?.text ?? ''
+                );
+            }
+        }
+        workspace.applyEdit(edit);
```

Comment:
> workspace.applyEdit(edit); [](start = 7, length = 27)

How does applying this edit affect the other "replacements" in the SARIF log? Do they need to be adjusted somehow?

Reply:
Yes they would need to be adjusted. This touches on my leading PR comment "There are certain scenarios (such as cascading fixes) that still require some discussion." We should discuss this at our next meeting.